monitoring_dashboard.qmd

```diff
@@ -0,0 +1,138 @@
+---
+title: "Monitoring dashboard"
+format:
+  dashboard:
+    orientation: columns
+logo: https://github.com/rstudio/vetiver-python/blob/main/docs/figures/logo.png?raw=true
+output: asis
+jupyter: python3
+---
+
+```{python}
+#| include: false
+#| tags: [parameters]
+
+# import model and metadata
+import pins
+from IPython.display import display, Markdown, IFrame
+from datetime import datetime, timedelta
+import pandas as pd
+import plotly.express as px
+from sklearn import metrics
+from vetiver import VetiverModel, compute_metrics, plot_metrics
+from sklearn.metrics import recall_score, accuracy_score
+
+raw = "https://colorado.rstudio.com/rsc"
+paths = {"chicago-model-python": "chicago-model-python/"}
+board = pins.board_url(raw, paths, allow_pickle_read=True)
+v = VetiverModel.from_pin(board, "chicago-model-python")
+v_meta = board.pin_meta("chicago-model-python")
+days_old = datetime.today() - datetime.strptime(v_meta.created, "%Y%m%dT%H%M%SZ")
+```
+
+```{python}
+## the next few lines are an example model, here is a place to
+## add any code you need to import new data and make predictions
+
+# import new data to track performance over time
+raw = "https://colorado.rstudio.com/rsc"
+paths = {"new-data": "inspections-new-data/"}
+board = pins.board_url(raw, paths, allow_pickle_read=True)
+inspections_new = board.pin_read("new-data")
+
+# make predictions
+inspections_new["preds"] = v.model.predict(
+    inspections_new.drop(columns=["results", "aka_name", "inspection_date"])
+)
+
+# map results
+inspections_new["preds"] = inspections_new["preds"].map({"PASS": 0, "FAIL": 1})
+inspections_new["results"] = inspections_new["results"].map({"PASS": 0, "FAIL": 1})
+```
+
+# Model card
```

Comment:
I hadn't really thought about putting the Model Card content in the monitoring dashboard template itself (vs. the example for the Quarto gallery). On the one hand, it's a good way to surface this and I suspect people may in fact find this a good way to go (one piece of published content with model info). On the other hand, this code is now in two places and we'd have to keep this updated in both, and this template now takes more work to get customized and is not just about monitoring.

What are your thoughts on this? Do you have a strong sense that we should include the Model Card info within the monitoring template?

Reply:
Hmm...maybe we don't have a heading that says `# Model Card`, but just include some of the information? It does seem nice to have a little bit of context, but it is a valid point that people don't want to maintain this information (or that we don't want to maintain this code 😂) in two locations. 

What do you think about keeping the automatable pieces that are useful for context/people won't have to customize, and then pull pretty much everything else? Concretely, we could keep:

- `v.description`
- `v.model_name`
- Printing the input prototype
- Prompting for "The model was created by ..."

as the index page of the dashboard.

Reply:
I think that's a good way to go. 👍